src/uct/ib/ud/base/ud_ep.c

```diff
@@ -552,7 +607,7 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         uct_ud_ep_rx_creq(iface, neth);
         goto out;
     } else if (ucs_unlikely(!ucs_ptr_array_lookup(&iface->eps, dest_id, ep) ||
-               ep->ep_id != dest_id))
+               (ep->ep_id != dest_id) || (ep->flags & UCT_UD_EP_FLAG_FAILED)))
```

Comment:
i see. well i guess this will be resolved when we would be able to remove from ptr_array immediately.